### PR TITLE
Fix: Glob pattern from .eslintignore not applied

### LINKED ISSUE #3746: 1.4.0 is linting ignored files
ESLint just upgraded and now our Travis builds are failing because it's linting an ignored file.

Travis build: https://travis-ci.org/discourse/discourse/jobs/79870981
It runs `eslint app/assets/javascripts` from the project directory.

The `.eslintignore` file is in the project directory: https://github.com/discourse/discourse/blob/22844b9e4663a1dd06040cd5913f02f571f9672f/.eslintignore and contains:
```
...
app/assets/javascripts/locales/i18n.js
app/assets/javascripts/defer/html-sanitizer-bundle.js
app/assets/javascripts/discourse/lib/Markdown.Editor.js
...
```

But the build outputs:

```
/home/travis/build/discourse/discourse/app/assets/javascripts/defer/html-sanitizer-bundle.js
  1243:14  error  Octal literal in strict mode
```

This happens on my local machine too.
```
$ eslint --version
v1.4.0
```

This just started happening with the new release (a few minutes ago).

### LINKED ISSUE #3748: jsdoc validation breaks
Running version 1.4.0 with `valid-jsdoc` set to `2`. We don't have any jsdoc comments in the codebase being linted.

```
/home/ubuntu/smart-rain-barrel/barrel-web/node_modules/eslint/lib/rules/valid-jsdoc.js:124
                        } else if (tag.name.indexOf(".") === -1) {
                                           ^
TypeError: Cannot read property 'indexOf' of undefined
    at /home/ubuntu/smart-rain-barrel/barrel-web/node_modules/eslint/lib/rules/valid-jsdoc.js:124:44
    at Array.forEach (native)
    at EventEmitter.checkJSDoc (/home/ubuntu/smart-rain-barrel/barrel-web/node_modules/eslint/lib/rules/valid-jsdoc.js:107:24)
    at EventEmitter.emit (events.js:129:20)
    at Controller.controller.traverse.leave (/home/ubuntu/smart-rain-barrel/barrel-web/node_modules/eslint/lib/eslint.js:834:25)
    at Controller.__execute (/home/ubuntu/smart-rain-barrel/barrel-web/node_modules/eslint/node_modules/estraverse/estraverse.js:397:31)
    at Controller.traverse (/home/ubuntu/smart-rain-barrel/barrel-web/node_modules/eslint/node_modules/estraverse/estraverse.js:485:28)
    at EventEmitter.module.exports.api.verify (/home/ubuntu/smart-rain-barrel/barrel-web/node_modules/eslint/lib/eslint.js:819:24)
    at processText (/home/ubuntu/smart-rain-barrel/barrel-web/node_modules/eslint/lib/cli-engine.js:217:27)
    at processFile (/home/ubuntu/smart-rain-barrel/barrel-web/node_modules/eslint/lib/cli-engine.js:254:18)
    at executeOnFile (/home/ubuntu/smart-rain-barrel/barrel-web/node_modules/eslint/lib/cli-engine.js:539:23)
```

### LINKED ISSUE #3750: Glob pattern from .eslintignore not applied
My `.eslintignore`:
```js
node_modules/**
test/*/**/*.js
```
In ESLint 1.3.x it was ignoring files properly; in 1.4.0 the second pattern doesn't seem to work; files like `test/npm-fixtures/not-ok-install-copy/node_modules/json3/lib/json3.js` are linted.

You can test it by checking out the `refactor` branch of my [check-dependencies](https://github.com/mzgol/check-dependencies) repo and invoking `npm install && npm test && npm test`. It hangs for a **long** time on ESLint and finally spits out lots of errors.

### PR DESCRIPTION
**Pull request by BYK**
Fixes #3750
Fixes #3746
Fixes #3748

### CODE REVIEW

**Comment by ilyavolodin** [comment]
LGTM

File: tests/lib/cli-engine.js [hunk 1]
```diff
@@ -460,6 +460,24 @@ describe("CLIEngine", function() {
...
+                ignorePath: getFixturePath(".eslintignore")
+            });
+
+            var report = engine.executeOnFiles(["./tests/fixtures/"]);
+            assert.equal(report.results.length, 0);
+        });
+
+        it("should return zero messages when all given files are ignored via ignore-pattern", function() {
+            engine = new CLIEngine({
+                ignorePattern: "tests/fixtures/single-quoted.js"
```
Comment (gyandeeps) [inline tests/lib/cli-engine.js]: `ignorePattern` ??

File: tests/lib/cli-engine.js [hunk 2]
```diff
@@ -460,6 +460,24 @@ describe("CLIEngine", function() {
             assert.equal(report.results.length, 0);
         });
 
+        it("should return zero messages when all given files are ignored event with a `./` prefix", function() {
```
Comment (platinumazure) [inline tests/lib/cli-engine.js]: "event with"? Should that be "even with"?

Comment (BYK) [inline tests/lib/cli-engine.js]: Yup, good catch :)

**Pull request merged**